### DISCUSSION TITLE
Monitor by process name

### DISCUSSION THREAD

**Discussion by zhangshenghang** [apache/hertzbeat#1851] [General]
Is it necessary to support monitoring whether the process is normal based on the process name? There are many scenarios where the services I start do not provide external ports and JMX monitoring.

**Comment by tomsun28** [comment]
👍👍 hi, feel like we can create a new monitoring type, Linux Process, which can be judged and implemented by running a shell script through the ssh protocol. Referring to the existing Linux monitoring logic, theoretically just write a shell script in the template YML.
